polaris/ocean/model/ocean_model_step.py

```diff
@@ -214,3 +258,148 @@ def _update_ntasks(self):
+            except ValueError:
+                not_found.append(mpaso_option)
+
+        self._warn_not_found(not_found)
+
+        return out_options
+
+    def _map_mpaso_to_omega_option(self, option, value):
+        """
+        Map MPAS-Ocean namelist option to Omega equivalent
+        """
+        out_option = option
+        found = False
+
+        assert self.map is not None
+        # traverse the map
+        for entry in self.map:
+            options_dict = entry['options']
+            for mpaso_option, omega_option in options_dict.items():
+                if option == mpaso_option:
+                    found = True
+                    out_option = omega_option
+                    break
+
+            if found:
+                break
+
+        if not found:
+            raise ValueError(f'No mapping found for {option}')
+
+        out_option, out_value = self._map_handle_not(out_option, value)
+
+        return out_option, out_value
+
+    def _map_mpaso_to_omega_configs(self, configs):
+        """
+        Map MPAS-Ocean namelist options to Omega config options
+        """
+        out_configs: Dict[str, Dict[str, str]] = {}
+        not_found = []
+        for section, options in configs.items():
+            for option, mpaso_value in options.items():
+                try:
+                    omega_section, omega_option, omega_value = \
+                        self._map_mpaso_to_omega_section_option(
+                            section=section, option=option, value=mpaso_value)
+                    if omega_section not in out_configs:
+                        out_configs[omega_section] = {}
+                    out_configs[omega_section][omega_option] = omega_value
+                except ValueError:
+                    not_found.append(f'{section}/{option}')
+
+        self._warn_not_found(not_found)
+
+        return out_configs
+
+    def _map_mpaso_to_omega_section_option(self, section, option, value):
+        """
+        Map MPAS-Ocean namelist section and option to Omega equivalent
+        """
+        out_section = section
+        out_option = option
+
+        assert self.map is not None
+
+        section_found = False
+        option_found = False
+
+        # traverse the map
+        for entry in self.map:
+            section_found = False
+            section_dict = entry['section']
+            for mpaso_section, omega_seciton in section_dict.items():
+                if section == mpaso_section:
+                    section_found = True
+                    break
+
+            if not section_found:
+                continue
+
+            options_dict = entry['options']
+            for mpaso_option, omega_option in options_dict.items():
+                if option == mpaso_option:
+                    option_found = True
+                    break
+
+            if option_found:
+                out_section = omega_seciton
+                out_option = omega_option
+                break
```

Comment:
I tested this version for the current config options for omega and a config option that doesn't map and it seems to perform fine
```suggestion
        option_found = False

        # traverse the map
        for entry in self.map:
            section_dict = entry['section']
            try:
                omega_section = section_dict[section]
            except KeyError:
                continue
            else:
                options_dict = entry['options']
                option_found = False
                try:
                    omega_option = options_dict[option]
                except KeyError:
                    continue
                else:
                    option_found = True
                    out_section = omega_section
                    out_option = omega_option
                    break
```

Reply:
Okay, I made this suggested change.  Thanks @cbegeman.  I'll test it again once https://github.com/E3SM-Project/Omega/pull/123 has been merged and the Omega submodule has been update.